ui/src/components/admin/ModelsManager/ModelManagerController.js

```diff
@@ -115,350 +120,356 @@ Model table state transitions for each action:
+                            ...importProgress,
+                            running: true,
+                          });
+                        }
+                      })
+                      .catch(async error => {
+                        await appendNotification({
+                          type: NOTIFICATION_TYPES.ERROR,
+                          message: 'Bulk Import of Research Somatic Variants Failed.',
+                          details: error.response
+                            ? error.response.data.error.message
+                            : error.message,
+                          timeout: false,
+                        });
                       });
-                    }
-                  })
-                  .catch(async error => {
-                    await appendNotification({
-                      type: NOTIFICATION_TYPES.ERROR,
-                      message: 'Bulk Import of Research Somatic Variants Failed.',
-                      details: error.response ? error.response.data.error.message : error.message,
+                  },
+                  bulkPublish: async () => {
+                    // display a temporary notification immediately while waiting for the server response
+                    const tempNotification = await appendNotification({
+                      type: NOTIFICATION_TYPES.LOADING,
+                      message: 'Initiating Bulk Publish...',
+                      details:
+                        'You can continue to use the CMS and will be notified when the publish is complete. Please note that some functions are unavailable during the publish process.',
                       timeout: false,
                     });
-                  });
-              },
-              bulkPublish: bulkActionCreator({
-                action: 'publish',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              bulkUnpublish: bulkActionCreator({
-                action: 'unpublish',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              bulkDelete: bulkActionCreator({
-                action: 'delete',
-                baseUrl,
-                state,
-                setState,
-                appendNotification,
-              }),
-              publishOne: async name => {
-                // Set loading true (lock UI)
-                await setState({
-                  isLoading: true,
-                });
-
-                singleAction('publish', name)
-                  .then(() => loadData(baseUrl, state))
-                  .then(async ([dataResponse, countResponse]) => {
+                    await publishBulk(state.selection)
+                      .then(async response => {
+                        if (response.data.success) {
+                          await setPublishProgress({
```

Comment:
If bulk publish is initiated successfully, we set `running` in the UI publishProgress to `true` which triggers the polling of publish progress to start (and displays the "Publishing:..." progress banner)